### PR TITLE
Add custom share link to stuff you need

### LINKED ISSUE #6583: Display unique share link in campaign template
#### User Story

As a user, when I finish registering to be an organ donor, I want to be able to access my social share links if I revisit the campaign page. 
#### Additional Information (optional)

This should display in the "stuff you need" section, if the share link is enabled for that campaign
#### Why This Matters

because Adam said so.

### PR DESCRIPTION
**Pull request by katiecrane**
#### What's this PR do?

Adds the user's custom share link into the "Stuff You Need" section if the campaign is setup to register organ donors.
#### How should this be reviewed?

Go to a campaign page for an organ donor campaign and make sure the link is there. Go to a non organ donor campaign and make sure the link is not there.
#### Relevant tickets

Fixes #6583
#### Checklist
- [ ] Documentation added for new features/changed endpoints.

### CODE REVIEW

File: lib/modules/dosomething/dosomething_campaign/dosomething_campaign.theme.inc [hunk 1]
```diff
@@ -204,10 +204,19 @@ function dosomething_campaign_preprocess_action_page(&$vars, &$wrapper) {
   // Plan.
 
   // 1) Materials
+
+  // Build the share link here in case registering organ donors
+  $campaign_path = url(current_path(), array('absolute' => TRUE));
+  $custom_organ_share_link = $campaign_path . '?source=user/' . $vars['user']->uid;
```
Comment (angaither) [inline lib/modules/dosomething/dosomething_campaign/dosomething_campaign.theme.inc]: you can pass the `?source=user` in as an `$options` array with the [url](https://api.drupal.org/api/drupal/includes%21common.inc/function/url/7.x) function

File: lib/modules/dosomething/dosomething_campaign/dosomething_campaign.theme.inc [hunk 2]
```diff
@@ -204,10 +204,19 @@ function dosomething_campaign_preprocess_action_page(&$vars, &$wrapper) {
...
+
   if (isset($campaign->items_needed)) {
     $materials['category'] = 'materials';
     $materials['title'] = t('Stuff You Need');
     $materials['content'] = $campaign->items_needed;
+    // if organ donation campaign, add in the share link here
+    if ($campaign->variables['register_organ_donors']) {
+      $materials['content'] .= '<ul><li><a href=' . $custom_organ_share_link . '>Your Custom Share Link</a></li></ul>';
```
Comment (angaither) [inline lib/modules/dosomething/dosomething_campaign/dosomething_campaign.theme.inc]: we probably want to give this another title 😏  but maybe @ngjo knows what we should call it

Comment (sbsmith86) [inline lib/modules/dosomething/dosomething_campaign/dosomething_campaign.theme.inc]: I feel like we shouldn't hyperlink these...as we only want people to copy and paste the link to share. If we hyperlink it, they have to right click/copy the link or click the link and basically go to the same page they are already on. Could we just add the full link in some `<code></code>` tags?

Comment (katiecrane) [inline lib/modules/dosomething/dosomething_campaign/dosomething_campaign.theme.inc]: @sbsmith86 oh yeah we could definitely just have the plain link there. That seems to make more sense after the process you described.

**Comment by ngjo** [comment]
Text could be something like:

_Copy and paste your custom link to invite friends and family to become an organ donor: <link>_ 

And not sure if we can get wordier (will probably run this by...Adam and Ben?) and more explicit so they understand why they should use the unique link:

_Copy and paste your custom link to invite friends and family to become an organ donor: <link> By using your custom link and starting the conversation around organ donation, you'll enter for the chance to win a $$$ scholarship!_

**Comment by katiecrane** [comment]
@ngjo @angaither @sbsmith86 it looks like this now:
image

**Comment by sbsmith86** [comment]
looks good to me.

**Comment by ngjo** [comment]
agreed!

**Comment by sbsmith86** [comment]
👍

**Pull request merged**